tests/integration/test_forwarding.py

```diff
@@ -40,16 +47,14 @@ def test():
     else:
         payload = data
 
+    if compress_response:
+        headers["Accept-Encoding"] = "gzip"
+
     response = relay.post(
         "/api/test/reflect", data=payload, headers=headers, stream=True
     )
     response.raise_for_status()
-    if compress_response:
```

Comment:
Had to undo this test restriction since reqwest decides on its own whether to re-compress the body now (since there's no other way)